modules/ingester/instance.go

```diff
@@ -154,13 +172,49 @@ func newInstance(instanceID string, limiter *Limiter, overrides ingesterOverride
 	return i, nil
 }
 
-func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) error {
+func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) *tempopb.PushResponse {
+	var errors util.MultiError
+
+	maxLiveErrorTraces := make([]int32, 0)
+	traceTooLargeErrorTraces := make([]int32, 0)
+	totalTracesDiscarded := 0
+
 	for j := range req.Traces {
+		index := int32(j)
+
 		err := i.PushBytes(ctx, req.Ids[j].Slice, req.Traces[j].Slice)
 		if err != nil {
-			return err
+			errorMessage := err.Error()
+			totalTracesDiscarded++
+
+			if _, ok := err.(*maxLiveTracesError); ok {
```

Comment:
let's use `errors.Is()`. it will handle error wrapping and is the canonical method of detecting error equality in go